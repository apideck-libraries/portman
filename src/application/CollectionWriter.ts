import { CollectionDefinition } from 'postman-collection'
import {
  injectEnvVariables,
  orderCollectionRequests,
  overwriteCollectionKeyValues,
  overwriteCollectionValues,
  overwriteCollectionSecurityValues,
  writeCollectionPreRequestScripts,
  writeRawReplacements
} from '.'
import { GlobalConfig, PortmanConfig, PortmanOptions } from '../types'
import { isEmptyObject } from 'utils'

export class CollectionWriter {
  public collection: CollectionDefinition
  config: PortmanConfig
  options: PortmanOptions

  constructor(config: PortmanConfig, options: PortmanOptions, collection: CollectionDefinition) {
    this.config = config
    this.options = options
    this.collection = collection
  }

  public execute(): void {
    const { envFile, baseUrl } = this.options

    const { globals = {} } = this.config
    const {
      collectionPreRequestScripts = [],
      securityOverwrites = {},
      keyValueReplacements = {},
      valueReplacements = {},
      rawReplacements = [],
      orderOfOperations = []
    } = globals as GlobalConfig

    let collection = this.collection

    // --- Portman - Set Security values for Postman
    if (securityOverwrites && !isEmptyObject(securityOverwrites)) {
      collection = overwriteCollectionSecurityValues(collection, securityOverwrites)
    }

    // --- Portman - Search for keys in dictionary to set the value if key is found anywhere in collection
    if (keyValueReplacements && !isEmptyObject(keyValueReplacements)) {
      collection = overwriteCollectionKeyValues(collection, keyValueReplacements)
    }

    // --- Portman - Search for keys in dictionary to set the values if value is found anywhere in collection
    if (valueReplacements && !isEmptyObject(valueReplacements)) {
      collection = overwriteCollectionValues(collection, valueReplacements)
    }

    // --- Portman - Use .env to inject variables on Collection
    if (envFile) {
      collection = injectEnvVariables(collection, envFile, baseUrl)
    }

    // --- Portman - Set manually order Postman requests
    if (orderOfOperations && orderOfOperations.length > 0) {
      collection = orderCollectionRequests(collection, orderOfOperations)
    }

    // --- Portman - Set Postman pre-requests
    if (collectionPreRequestScripts && collectionPreRequestScripts.length > 0) {
      collection = writeCollectionPreRequestScripts(collection, collectionPreRequestScripts)
    }

    // --- Portman - Replace & clean-up Postman
    if (rawReplacements && rawReplacements.length > 0) {
      const collectionString = writeRawReplacements(
        JSON.stringify(collection, null, 2),
        rawReplacements
      )
      this.collection = JSON.parse(collectionString)
    } else {
      this.collection = collection
    }
  }
}
