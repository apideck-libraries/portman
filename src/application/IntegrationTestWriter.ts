import { Collection, Item, ItemGroup } from 'postman-collection'
import { IntegrationTestConfig, VariationConfig, PortmanTestTypes } from '../types'
import { TestSuite, VariationWriter } from './'

export class IntegrationTestWriter {
  public testSuite: TestSuite
  integrationTestFolder: ItemGroup<Item>
  integrationTestCollection: Collection

  constructor({
    testSuite,
    integrationTestFolderName
  }: {
    testSuite: TestSuite
    integrationTestFolderName: string
  }) {
    this.testSuite = testSuite
    this.integrationTestFolder = new ItemGroup<Item>({
      name: integrationTestFolderName
    })
    this.integrationTestCollection = new Collection()
  }

  public add(integrationTest: IntegrationTestConfig): void {
    const { testSuite } = this
    const { name, operations } = integrationTest

    const variationWriter = new VariationWriter({
      testSuite: this.testSuite,
      variationFolderName: name
    })

    operations.map(operation => {
      const variations = (operation?.variations || []) as VariationConfig[]
      const { openApiOperationId } = operation

      const pmOperation = testSuite.postmanParser.getOperationById(openApiOperationId)

      if (!pmOperation) return

      const oaOperation = testSuite.oasParser.getOperationByPath(pmOperation.pathRef)

      // Handle integration variations
      variations.map(variation => {
        const variationName = variation.name

        const operationVariation = pmOperation.clone({
          newId: `${variationName}-${Math.random()}`,
          name: variationName
        })

        // Set/Update Portman operation test type
        this.testSuite.registerOperationTestType(operationVariation, PortmanTestTypes.integration)

        variationWriter.injectVariations(operationVariation, oaOperation, variation, operation)
        variationWriter.addToFolder(operationVariation, variationWriter.variationFolder)
      })
    })

    this.integrationTestCollection = variationWriter.mergeToCollection(
      this.integrationTestCollection
    )
  }

  public mergeToCollection(collection: Collection): Collection {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    this.integrationTestCollection.items.map((item: any) => {
      this.integrationTestFolder.items.add(item)
    })

    collection.items.add(this.integrationTestFolder)
    return collection
  }
}
