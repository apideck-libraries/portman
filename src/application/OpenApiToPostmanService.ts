import fs from 'fs-extra'
import oaConverter from 'openapi-to-postmanv2'
import ora from 'ora'
import path from 'path'

export class OpenApiToPostmanService {
  /**
   * Helper function for the CLI to convert OpenApi data input
   * (ported from https://github.com/postmanlabs/openapi-to-postman)
   * @param {String} openApiObj - OpenApi spec file used for conversion input
   * @param options - OpenApi to Postman conversion options
   * @returns {string}
   */
  // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
  async convert(openApiObj: string, options: any): Promise<string> {
    return await new Promise((resolve, reject) => {
      const inputFile = path.resolve(options.inputFile)
      openApiObj = fs.readFileSync(inputFile, 'utf8')

      // apply options from config file if present
      if (options.configFile) {
        const configFile = path.resolve(options.configFile)
        // console.log('Options Config file: ', configFile)
        const configFileOptions = JSON.parse(fs.readFileSync(configFile, 'utf8'))
        options = Object.assign({}, options, configFileOptions)
      }

      if (options.testsuiteFile) {
        const testsuiteFile = path.resolve(options.testsuiteFile)
        // console.log('Testsuite file: ', testsuiteFile)
        options.testSuite = true
        options.testSuiteSettings = JSON.parse(fs.readFileSync(testsuiteFile, 'utf8'))
      }

      const spinner = ora({
        prefixText: ' ',
        text: 'Converting OpenApi to Postman Collection'
      }).start()

      // Convert OpenApi to Postman collection
      oaConverter.convert(
        {
          type: 'string',
          data: openApiObj
        },
        options,
        (err, status) => {
          if (err) {
            spinner.fail(err.toString())
            reject(err)
          }
          if (!status.result) {
            // console.log(status.reason)
            spinner.fail(status.reason)
            reject(status.reason)
          } else if (options.outputFile) {
            const filePath = path.resolve(options.outputFile)

            try {
              fs.writeFileSync(filePath, JSON.stringify(status.output[0].data, null, 4))
            } catch (error) {
              console.log('Could not write to file', error)
              spinner.fail(error.toString())
              reject(error)
            }

            // Return Postman collection
            spinner.succeed('Conversion successful')
            resolve(status.output[0].data)
          } else {
            spinner.succeed('Conversion successful')
            resolve(status.output[0].data)
          }
        }
      )
    })
  }
}
