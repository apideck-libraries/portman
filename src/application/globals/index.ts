export * from './injectEnvVariables'
export * from './orderCollectionRequests'
export * from './overwriteCollectionKeyValues'
export * from './overwriteCollectionSecurityValues'
export * from './overwriteCollectionValues'
export * from './renamePostmanCollection'
export * from './stripResponseExamples'
export * from './writeCollectionPreRequestScripts'
export * from './writeRawReplacements'
