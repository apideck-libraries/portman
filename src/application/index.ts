export * from './CollectionWriter'
export * from './globals'
export * from './newman'
export * from './overwrites'
export * from './tests'
export * from './TestSuite'
export * from './variables'
