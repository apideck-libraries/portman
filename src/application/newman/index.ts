export * from './runNewmanWith'
export * from './writeNewmanEnv'
