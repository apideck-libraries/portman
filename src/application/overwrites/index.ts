export * from './applyOverwrites'
export * from './overwriteRequestBody'
export * from './overwriteRequestHeaders'
export * from './overwriteRequestPathIdVariables'
export * from './overwriteRequestPathVariables'
export * from './overwriteRequestQueryParams'
export * from './overwriteRequestSecurity'
