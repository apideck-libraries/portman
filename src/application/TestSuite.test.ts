import { TestSuite } from 'application'
import fs from 'fs-extra'
import { getConfig } from 'lib'
import { OpenApiParser } from 'oas'
import { PostmanParser } from 'postman'

describe('TestSuite', () => {
  let postmanParser: PostmanParser
  let oasParser: OpenApiParser
  let testSuiteService: TestSuite

  const postmanJson = '__tests__/fixtures/crm.postman.json'
  const oasYml = '__tests__/fixtures/crm.yml'
  const postmanConfigFile = '__tests__/fixtures/portman.crm.json'

  beforeEach(async () => {
    oasParser = new OpenApiParser()
    await oasParser.convert({ inputFile: oasYml })
    const postmanObj = JSON.parse(fs.readFileSync(postmanJson).toString())
    const config = await getConfig(postmanConfigFile)
    postmanParser = new PostmanParser({ postmanObj: postmanObj, oasParser: oasParser })

    testSuiteService = new TestSuite({ oasParser, postmanParser, config })
  })

  it('should generateAutomatedTests', async () => {
    testSuiteService.generateAutomatedTests()
    // dig down to specific test as postman injects unique ids making snapshots  meh
    expect(postmanParser.mappedOperations[0].getTests().script.exec).toMatchSnapshot()
  })

  it('should injectContentTests', async () => {
    testSuiteService.injectContentTests()
    expect(postmanParser.mappedOperations[0].getTests().script.exec).toMatchSnapshot()
  })

  it('should injectAssignVariables', async () => {
    testSuiteService.injectAssignVariables()
    expect(postmanParser.mappedOperations[0].getTests().script.exec).toMatchSnapshot()
  })

  it('should injectExtendedTests', async () => {
    testSuiteService.injectExtendedTests()
    expect(postmanParser.mappedOperations[0].getTests().script.exec).toMatchSnapshot()
  })

  describe('injectOverwrites', () => {
    it('should overwriteRequestBody', async () => {
      testSuiteService.injectOverwrites()
      expect(postmanParser.mappedOperations[1].item.request.body).toMatchSnapshot()
    })

    it('should overwriteRequestQueryParams', async () => {
      testSuiteService.injectOverwrites()

      expect(postmanParser.mappedOperations[0].item.request.url.query).toMatchSnapshot()
    })

    it('should overwriteRequestPathVariables', async () => {
      testSuiteService.injectOverwrites()

      expect(postmanParser.mappedOperations[3].item.request.url.variables).toMatchSnapshot()
    })

    it('should overwriteRequestHeaders', async () => {
      testSuiteService.injectOverwrites()

      expect(postmanParser.mappedOperations[3].item.request.headers).toMatchSnapshot()
    })
  })
})
