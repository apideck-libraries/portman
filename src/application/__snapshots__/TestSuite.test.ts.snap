// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestSuite injectOverwrites should overwriteRequestBody 1`] = `
Object {
  "mode": "raw",
  "raw": "{
    \\"name\\": \\"Foo Inc. --{{$randomInt}}\\",
    \\"owner_id\\": \\"12345\\",
    \\"image_url\\": \\"https://www.spacex.com/static/images/share.jpg\\",
    \\"description\\": \\"A crm that works for you, so you can spend time on relationships instead of data.\\",
    \\"vat_number\\": \\"BE0689615164\\",
    \\"currency\\": \\"USD\\",
    \\"status\\": \\"Open\\",
    \\"fax\\": \\"+12129876543\\",
    \\"bank_accounts\\": [
        {
            \\"iban\\": \\"CH2989144532982975332\\",
            \\"bic\\": \\"AUDSCHGGXXX\\"
        },
        {
            \\"iban\\": \\"CH2989144532982975332\\",
            \\"bic\\": \\"AUDSCHGGXXX\\"
        }
    ],
    \\"websites\\": [
        {
            \\"url\\": \\"http://example.com\\",
            \\"id\\": \\"12345\\",
            \\"type\\": \\"primary\\"
        },
        {
            \\"url\\": \\"http://example.com\\",
            \\"id\\": \\"12345\\",
            \\"type\\": \\"primary\\"
        }
    ],
    \\"addresses\\": [
        {
            \\"id\\": \\"123\\",
            \\"type\\": \\"primary\\",
            \\"name\\": \\"HQ US\\",
            \\"line1\\": \\"Main street\\",
            \\"line2\\": \\"apt #\\",
            \\"city\\": \\"San Francisco\\",
            \\"state\\": \\"CA\\",
            \\"postal_code\\": \\"94104\\",
            \\"country\\": \\"US\\",
            \\"latitude\\": \\"40.759211\\",
            \\"longitude\\": \\"-73.984638\\"
        },
        {
            \\"id\\": \\"123\\",
            \\"type\\": \\"primary\\",
            \\"name\\": \\"HQ US\\",
            \\"line1\\": \\"Main street\\",
            \\"line2\\": \\"apt #\\",
            \\"city\\": \\"San Francisco\\",
            \\"state\\": \\"CA\\",
            \\"postal_code\\": \\"94104\\",
            \\"country\\": \\"US\\",
            \\"latitude\\": \\"40.759211\\",
            \\"longitude\\": \\"-73.984638\\"
        }
    ],
    \\"social_links\\": [
        {
            \\"url\\": \\"https://www.twitter.com/apideck-io\\",
            \\"id\\": \\"12345\\",
            \\"type\\": \\"twitter\\"
        },
        {
            \\"url\\": \\"https://www.twitter.com/apideck-io\\",
            \\"id\\": \\"12345\\",
            \\"type\\": \\"twitter\\"
        }
    ],
    \\"phone_numbers\\": [
        {
            \\"number\\": \\"111-111-1111\\",
            \\"id\\": \\"12345\\",
            \\"type\\": \\"primary\\"
        },
        {
            \\"number\\": \\"111-111-1111\\",
            \\"id\\": \\"12345\\",
            \\"type\\": \\"primary\\"
        }
    ],
    \\"emails\\": [
        {
            \\"email\\": \\"elon@musk.com\\",
            \\"id\\": \\"123\\",
            \\"type\\": \\"primary\\"
        },
        {
            \\"email\\": \\"elon@musk.com\\",
            \\"id\\": \\"123\\",
            \\"type\\": \\"primary\\"
        }
    ],
    \\"custom_fields\\": [
        {
            \\"id\\": \\"custom_technologies\\",
            \\"value\\": \\"Uses Salesforce and Marketo\\"
        },
        {
            \\"id\\": \\"custom_technologies\\",
            \\"value\\": \\"Uses Salesforce and Marketo\\"
        }
    ],
    \\"tags\\": [
        \\"New\\"
    ]
}",
}
`;

exports[`TestSuite injectOverwrites should overwriteRequestHeaders 1`] = `
Array [
  Object {
    "description": Object {
      "content": "(Required) ID of the consumer which you want to get or push data from",
      "type": "text/plain",
    },
    "disabled": false,
    "key": "x-apideck-consumer-id",
    "value": "portman-id-{{$randomInt}}",
  },
  Object {
    "description": Object {
      "content": "(Required) The ID of your Unify application",
      "type": "text/plain",
    },
    "disabled": false,
    "key": "x-apideck-app-id",
    "value": "2222",
  },
  Object {
    "description": Object {
      "content": "Provide the service id you want to call (e.g., pipedrive). [See the full list in the connector section.](#section/Connectors) Only needed when a consumer has activated multiple integrations for a Unified API.",
      "type": "text/plain",
    },
    "disabled": false,
    "key": "x-apideck-service-id",
    "value": "salesforce",
  },
  Object {
    "key": "Content-Type",
    "value": "application/json",
  },
]
`;

exports[`TestSuite injectOverwrites should overwriteRequestPathVariables 1`] = `
Array [
  Object {
    "description": Object {
      "content": "(Required) ID of the record you are acting upon.",
      "type": "text/plain",
    },
    "disabled": false,
    "key": "id",
    "type": "string",
    "value": "123456789",
  },
]
`;

exports[`TestSuite injectOverwrites should overwriteRequestQueryParams 1`] = `
Array [
  Object {
    "description": Object {
      "content": "Include raw response. Mostly used for debugging purposes",
      "type": "text/plain",
    },
    "disabled": false,
    "key": "raw",
    "value": "true",
  },
  Object {
    "description": Object {
      "content": "Cursor to start from. You can find cursors for next/previous pages in the meta.cursors property of the response.",
      "type": "text/plain",
    },
    "disabled": false,
    "key": "cursor",
    "value": "",
  },
  Object {
    "description": Object {
      "content": "Number of records to return",
      "type": "text/plain",
    },
    "disabled": false,
    "key": "limit",
    "value": "4",
  },
  Object {
    "description": Object {
      "content": "Apply filters (beta)",
      "type": "text/plain",
    },
    "disabled": false,
    "key": "filter[name]",
    "value": "SpaceX",
  },
]
`;

exports[`TestSuite should generateAutomatedTests 1`] = `
Array [
  "// Validate status 2xx 
pm.test(\\"[GET]::/crm/companies - Status code is 2xx\\", function () {
   pm.response.to.be.success;
});
",
  "// Validate response time 
pm.test(\\"[GET]::/crm/companies - Response time is less than 300ms\\", function () {
    pm.expect(pm.response.responseTime).to.be.below(300);
});
",
  "// Validate if response header has matching content-type
pm.test(\\"[GET]::/crm/companies - Content-Type is application/json\\", function () {
   pm.expect(pm.response.headers.get(\\"Content-Type\\")).to.include(\\"application/json\\");
});
",
  "// Validate if response has JSON Body 
pm.test(\\"[GET]::/crm/companies - Response has JSON Body\\", function () {
    pm.response.to.have.jsonBody();});
",
  "// Response Validation
const schema = {\\"x-graphql-type-name\\":\\"CompanyList\\",\\"type\\":\\"object\\",\\"required\\":[\\"status_code\\",\\"status\\",\\"service\\",\\"resource\\",\\"operation\\",\\"data\\"],\\"properties\\":{\\"status_code\\":{\\"type\\":\\"integer\\",\\"description\\":\\"HTTP Response Status Code\\",\\"example\\":200},\\"status\\":{\\"type\\":\\"string\\",\\"description\\":\\"HTTP Response Status\\",\\"example\\":\\"OK\\"},\\"service\\":{\\"type\\":\\"string\\",\\"description\\":\\"Apideck ID of service provider\\",\\"example\\":\\"zoho-crm\\"},\\"resource\\":{\\"type\\":\\"string\\",\\"description\\":\\"Unified API resource name\\",\\"example\\":\\"companies\\"},\\"operation\\":{\\"type\\":\\"string\\",\\"description\\":\\"Operation performed\\",\\"example\\":\\"one\\"},\\"data\\":{\\"type\\":\\"array\\",\\"items\\":{\\"type\\":\\"object\\",\\"required\\":[\\"name\\"],\\"additionalProperties\\":false,\\"properties\\":{\\"id\\":{\\"type\\":\\"string\\",\\"readOnly\\":true,\\"example\\":\\"12345\\"},\\"interaction_count\\":{\\"type\\":[\\"integer\\",\\"null\\"],\\"example\\":1,\\"readOnly\\":true},\\"name\\":{\\"type\\":\\"string\\",\\"example\\":\\"Copper\\"},\\"owner_id\\":{\\"type\\":\\"string\\",\\"example\\":\\"12345\\"},\\"image_url\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"https://www.spacex.com/static/images/share.jpg\\"},\\"description\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"A crm that works for you, so you can spend time on relationships instead of data.\\"},\\"vat_number\\":{\\"description\\":\\"VAT number\\",\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"BE0689615164\\"},\\"currency\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"USD\\"},\\"status\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"minLength\\":1,\\"example\\":\\"Open\\"},\\"fax\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"+12129876543\\"},\\"bank_accounts\\":{\\"type\\":\\"array\\",\\"items\\":{\\"type\\":\\"object\\",\\"properties\\":{\\"iban\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"CH2989144532982975332\\"},\\"bic\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"AUDSCHGGXXX\\"}}}},\\"websites\\":{\\"type\\":\\"array\\",\\"items\\":{\\"type\\":\\"object\\",\\"required\\":[\\"url\\"],\\"properties\\":{\\"id\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"12345\\"},\\"url\\":{\\"type\\":\\"string\\",\\"example\\":\\"http://example.com\\"},\\"type\\":{\\"type\\":\\"string\\",\\"x-graphql-type-name\\":\\"WebsiteType\\",\\"enum\\":[\\"primary\\",\\"secondary\\",\\"work\\",\\"personal\\",\\"other\\"],\\"example\\":\\"primary\\"}}}},\\"addresses\\":{\\"type\\":\\"array\\",\\"items\\":{\\"type\\":\\"object\\",\\"properties\\":{\\"id\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"123\\"},\\"type\\":{\\"type\\":\\"string\\",\\"x-graphql-type-name\\":\\"AddressType\\",\\"enum\\":[\\"primary\\",\\"secondary\\",\\"home\\",\\"office\\",\\"shipping\\",\\"billing\\",\\"other\\"],\\"example\\":\\"primary\\"},\\"name\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"HQ US\\"},\\"line1\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"Main street\\",\\"description\\":\\"Line 1 of the address e.g. number, street, suite, apt #, etc.\\"},\\"line2\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"apt #\\",\\"description\\":\\"Line 2 of the address\\"},\\"city\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"San Francisco\\",\\"description\\":\\"Name of city.\\"},\\"state\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"CA\\",\\"description\\":\\"Name of state\\"},\\"postal_code\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"94104\\",\\"description\\":\\"Zip code or equivalent.\\"},\\"country\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"US\\",\\"description\\":\\"country code according to ISO 3166-1 alpha-2.\\"},\\"latitude\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"40.759211\\"},\\"longitude\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"-73.984638\\"}}}},\\"social_links\\":{\\"type\\":\\"array\\",\\"items\\":{\\"required\\":[\\"url\\"],\\"type\\":\\"object\\",\\"properties\\":{\\"id\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"12345\\"},\\"url\\":{\\"type\\":\\"string\\",\\"example\\":\\"https://www.twitter.com/apideck-io\\"},\\"type\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"twitter\\"}}}},\\"phone_numbers\\":{\\"type\\":\\"array\\",\\"items\\":{\\"required\\":[\\"number\\"],\\"type\\":\\"object\\",\\"properties\\":{\\"id\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"12345\\"},\\"number\\":{\\"type\\":\\"string\\",\\"example\\":\\"111-111-1111\\"},\\"type\\":{\\"type\\":\\"string\\",\\"x-graphql-type-name\\":\\"PhoneType\\",\\"enum\\":[\\"primary\\",\\"secondary\\",\\"home\\",\\"office\\",\\"mobile\\",\\"assistant\\",\\"fax\\",\\"other\\"],\\"example\\":\\"primary\\"}}}},\\"emails\\":{\\"type\\":\\"array\\",\\"items\\":{\\"required\\":[\\"email\\"],\\"type\\":\\"object\\",\\"properties\\":{\\"id\\":{\\"type\\":\\"string\\",\\"example\\":\\"123\\"},\\"email\\":{\\"type\\":\\"string\\",\\"format\\":\\"email\\",\\"example\\":\\"elon@musk.com\\"},\\"type\\":{\\"type\\":\\"string\\",\\"x-graphql-type-name\\":\\"EmailType\\",\\"enum\\":[\\"primary\\",\\"secondary\\",\\"work\\",\\"personal\\",\\"billing\\",\\"other\\"],\\"example\\":\\"primary\\"}}}},\\"custom_fields\\":{\\"type\\":\\"array\\",\\"items\\":{\\"type\\":\\"object\\",\\"required\\":[\\"id\\"],\\"additionalProperties\\":false,\\"properties\\":{\\"id\\":{\\"type\\":\\"string\\",\\"example\\":\\"custom_technologies\\"},\\"value\\":{\\"anyOf\\":[{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"Uses Salesforce and Marketo\\"},{\\"type\\":[\\"number\\",\\"null\\"],\\"example\\":10},{\\"type\\":[\\"boolean\\",\\"null\\"],\\"example\\":true},{\\"type\\":\\"array\\",\\"items\\":{\\"type\\":\\"string\\"}}]}}}},\\"tags\\":{\\"type\\":\\"array\\",\\"items\\":{\\"type\\":\\"string\\"},\\"example\\":[\\"New\\"]},\\"updated_by\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"12345\\",\\"readOnly\\":true},\\"created_by\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"example\\":\\"12345\\",\\"readOnly\\":true},\\"updated_at\\":{\\"type\\":\\"string\\",\\"example\\":\\"2020-09-30T07:43:32.000Z\\",\\"readOnly\\":true},\\"created_at\\":{\\"type\\":\\"string\\",\\"example\\":\\"2020-09-30T07:43:32.000Z\\",\\"readOnly\\":true}}}},\\"meta\\":{\\"type\\":\\"object\\",\\"description\\":\\"Reponse metadata\\",\\"properties\\":{\\"items_on_page\\":{\\"type\\":\\"integer\\",\\"description\\":\\"Number of items returned in the data property of the response\\",\\"example\\":50},\\"cursors\\":{\\"type\\":\\"object\\",\\"description\\":\\"Cursors to navigate to previous or next pages through the API\\",\\"properties\\":{\\"previous\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"description\\":\\"Cursor to navigate to the previous page of results through the API\\",\\"example\\":\\"em9oby1jcm06OnBhZ2U6OjE=\\"},\\"current\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"description\\":\\"Cursor to navigate to the current page of results through the API\\",\\"example\\":\\"em9oby1jcm06OnBhZ2U6OjI=\\"},\\"next\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"description\\":\\"Cursor to navigate to the next page of results through the API\\",\\"example\\":\\"em9oby1jcm06OnBhZ2U6OjM=\\"}}}}},\\"links\\":{\\"type\\":\\"object\\",\\"description\\":\\"Links to navigate to previous or next pages through the API\\",\\"properties\\":{\\"previous\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"description\\":\\"Link to navigate to the previous page through the API\\",\\"example\\":\\"https://unify.apideck.com/crm/companies?cursor=em9oby1jcm06OnBhZ2U6OjE%3D\\"},\\"current\\":{\\"type\\":\\"string\\",\\"description\\":\\"Link to navigate to the current page through the API\\",\\"example\\":\\"https://unify.apideck.com/crm/companies\\"},\\"next\\":{\\"type\\":[\\"string\\",\\"null\\"],\\"description\\":\\"Link to navigate to the previous page through the API\\",\\"example\\":\\"https://unify.apideck.com/crm/companies?cursor=em9oby1jcm06OnBhZ2U6OjM\\"}}}}}

// Validate if response matches JSON schema 
pm.test(\\"[GET]::/crm/companies - Schema is valid\\", function() {
    pm.response.to.have.jsonSchema(schema,{unknownFormats: [\\"int32\\", \\"int64\\"]});
});
",
]
`;

exports[`TestSuite should injectAssignVariables 1`] = `
Array [
  "// Set response object as internal variable
let jsonData = pm.response.json();
",
  "// pm.collectionVariables - Set companiesAll.data[0].id as variable for jsonData.data[0].id  
if (typeof jsonData.data[0].id !== \\"undefined\\") {
   pm.collectionVariables.set(\\"companiesAll.data[0].id\\", jsonData.data[0].id);
   console.log(\\"- use {{companiesAll.data[0].id}} as collection variable for value\\",jsonData.data[0].id);
};
",
]
`;

exports[`TestSuite should injectContentTests 1`] = `
Array [
  "// Set response object as internal variable
let jsonData = pm.response.json();
",
  "// Response body should have property \\"data[0].company_name\\"
pm.test(\\"[GET]::/crm/companies - Content check if property 'data[0].company_name' exists\\", function() {
   pm.expect((typeof jsonData.data[0].company_name !== \\"undefined\\")).to.be.true;
});
",
  "// Response body should have value \\"Spacex\\" for \\"data[0].company_name\\"
if (typeof jsonData.data[0].company_name !== \\"undefined\\") {
pm.test(\\"[GET]::/crm/companies - Content check if value for 'data[0].company_name' matches 'Spacex'\\", function() {
  pm.expect(jsonData.data[0].company_name).to.eql(\\"Spacex\\");
})};",
]
`;

exports[`TestSuite should injectExtendedTests 1`] = `
Array [
  "pm.test('[GET] /crm/companies - Unify pagination limit', function(){pm.expect(pm.response.json().data.length).to.eql(3);});",
]
`;
